src/ECharts/Types.purs

```diff
@@ -27,6 +36,26 @@ pixelOrPercentToForeign = case _ of
   Pixel i → toForeign i
   Percent n → toForeign $ show n <> "%"
 
+data VagueDate
+  = VagueYear Int
```

Comment:
we need to be able to specify '2016' from '2016-01' from '2016-01-30' though, since the API supports usages like the following:

* range year to yearmonth `['2017', '2017-05']`
* range year to yearmonthdate ['2017', '2017-05-15']
* range yearmonth to year `['2017-05-01', '2018']`
* reverse ordered range `['2017-05-01', '2017']`

i guess i could add a type for date resolution in any case